apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FirestoreDb.cs

```diff
@@ -155,6 +155,15 @@ internal DocumentReference GetDocumentReferenceFromResourceName(string reference
                 nameof(referenceValue), "Reference {0} is not part of database {1}", referenceValue, RootPath);
             return Document(referenceValue.Substring(DocumentsPath.Length + 1));
         }
+        
+        /// <summary>
+        /// Fetches document snapshots from the server, based on an optional transaction ID.
+        /// </summary>
+        /// <param name="documents">The document references to fetch. Must not be null, or contain null references.</param>
+        /// <param name="cancellationToken">A cancellation token for the operation.</param>
+        /// <returns>The document snapshots, in the order they are provided in the response. (This may not be the order of <paramref name="documents"/>.)</returns>
+        public Task<IList<DocumentSnapshot>> SnapshotAsync(IEnumerable<DocumentReference> documents, CancellationToken cancellationToken = default) =>
```

Comment:
I would advise not to worry too much about consistent naming between these methods, their Proto, auto-generated or internal counterparts. It's much more important to create a consistent API experience within Firestore and Google Cloud .Net. 
I am not fully sold on SnapshotAsync (rather than SnapshotsAsync), but I do like the consistency between DocumentReference, Query and this class.